src/components/AssertionForm/ActionButton/ActionButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { useBalanceAndAllowance } from "@/hooks";
+import type { ChainId } from "@/types";
+import type { Address } from "wagmi";
+import styles from "./ActionButton.module.css";
+import { ApproveButton } from "./ApproveButton";
+import { SubmitButton } from "./SubmitButton";
+
+export type Props = {
+  userAddress: Address;
+  oracleAddress: Address;
+  chainId: ChainId;
+  claim: string;
+  bondBigInt: bigint;
+  challengePeriodBigInt: bigint;
+  currencyAddress: Address;
+  decimals: number;
+};
+
+export function ActionButton(props: Props) {
+  const { hasApproved, insufficientFunds } = useActionButton(props);
```

Comment:
i do like this a lot better, the only thing i dont like here is that props need to be passed in redundantly. seems it would be better to `useActionButton` in the parent component and pass the result in as props to this button.

Reply:
the problem with this is that we want the props to not be undefined if the `ActionButton` renders. Otherwise we get all the odd behavior of passing undefined to these hooks. So really these props are technically different props, because the props in the parent component can be undefined.